src/View/SerializedView.php

```diff
@@ -84,8 +85,23 @@ public function render($view = null, $layout = null)
         }
 
         if ($serialize !== false) {
-            return $this->_serialize($serialize);
-        } elseif ($view !== false && $this->_getViewFileName($view)) {
+            $result = $this->_serialize($serialize);
+            if ($result === false) {
+                if (Configure::read('debug') && in_array('trace', $serialize)) {
+                    $keys = array_keys($serialize, 'trace');
+                    foreach ($keys as $key) {
+                        unset($serialize[$key]);
+                    }
+                    $result = $this->_serialize($serialize);
+                    if ($result !== false) {
+                        return $result;
+                    }
+                }
+                trigger_error('Serialization of View data failed.');
```

Comment:
This would not be too BC for those cases where it happens and people live with it currently.
But I would agree.